src/goSuggest.ts

```diff
@@ -55,7 +55,22 @@ export class GoCompletionItemProvider implements vscode.CompletionItemProvider {
 				// prevent completion when typing in a line comment
 				const commentIndex = lineText.indexOf('//');
 				if (commentIndex >= 0 && position.character > commentIndex) {
-					return resolve([]);
+					if (!lineText.trim().startsWith('//')) {
+						return resolve([]);
+					}
+					// triggering completions in comments on exported members
+					if (position.line + 1 < document.lineCount) {
+						let nextLine = document.lineAt(position.line + 1).text.trim();
+						let memberType = nextLine.match(/(const|func|type|var)\s*(\w+)/);
+						if (memberType && memberType.length === 3) {
```

Comment:
In the test file, at lines 11, 13, 18, 20 and 35 we get multiple completions from `gocode` and not from the code we specifically wrote for exported members.

This is because we don't exit with empty results when there is no match with the regex above.

Reply:
So in case of a block declaration, we should not trigger any completions and exit with empty results?

Reply:
Yes, unless you want to scan through the lines above and below the current line to look for the starting and ending of the block

Reply:
I think I am fine with skipping, wanted to clarify before I made any changes.